### PR TITLE
Minor cleanup in  apiserver startup in integration tests

### PR DESCRIPTION
**Pull request by wojtek-t**
```release-note
NONE
```

/kind cleanup
/priority important-longerm
/sig testing

### CODE REVIEW

File: test/integration/util/util.go [hunk 1]
```diff
@@ -68,13 +68,7 @@ type ShutdownFunc func()
...
 // StartApiserver starts a local API server for testing and returns the handle to the URL and the shutdown function to stop it.
 func StartApiserver() (string, ShutdownFunc) {
-	h := &framework.APIServerHolder{Initialized: make(chan struct{})}
-	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		<-h.Initialized
-		h.M.GenericAPIServer.Handler.ServeHTTP(w, req)
-	}))
-
-	_, _, closeFn := framework.RunAnAPIServerUsingServer(framework.NewIntegrationTestControlPlaneConfig(), s, h)
+	_, s, closeFn := framework.RunAnAPIServer(framework.NewIntegrationTestControlPlaneConfig())
```
Comment (aojea) [inline test/integration/util/util.go:71]: the CloseFn returned already has a s.Close()
https://github.com/kubernetes/kubernetes/blob/ce50018a760dda229377f29032492cfee94e52ec/test/integration/framework/master_utils.go#L127-L153

I think that the `shutdownFunc` can be removed with this change

Comment (wojtek-t) [reply test/integration/util/util.go:71]: Good catch - PTAL

**Comment by aojea** [comment]
> Kubernetes e2e suite: [sig-node] Downward API should provide pod UID as env vars [NodeConformance] [Conformance] expand_less	

I start to see this failures on Node tests more often :eyes: 
/retest

**Pull request merged**